From 1c5a71eca500e43418d9a7b2e0f5efc89c2716b2 Mon Sep 17 00:00:00 2001
From: "junwei.ma" <junwei.ma@amlogic.com>
Date: Thu, 20 Jul 2023 17:00:33 +0800
Subject: [PATCH] ISP: [isp][Kernel 5.15] output RGB not BGR [1/1]

PD#SWPL-112858

Problem:
rgb to bgr

Solution:
not convert anymore

Verify:
ok

Change-Id: Ib5c26ee60ba45569d35ed956c4f65714c50cbb37
Signed-off-by: junwei.ma <junwei.ma@amlogic.com>
---
 linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc.c  | 2 +-
 linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc1.c | 4 ++--
 linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc2.c | 2 +-
 linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc3.c | 2 +-
 4 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc.c b/linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc.c
index fd7486d0..a72f84aa 100644
--- a/linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc.c
+++ b/linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc.c
@@ -546,7 +546,7 @@ static void enable_isp_scale_new (
         sc_wr_reg_bits(ISP_SCWR_TOP_CTRL, 1, 19, 1);
     }
 
-    if ( mux_sel != 4  && wr_mif->reg_rgb_mode != 0 )
+    if ( mux_sel != 4  && wr_mif->reg_rgb_mode == 2 )
         sc_wr_reg_bits(ISP_SCWR_TOP_CTRL, 5, 13, 3);
     else
         sc_wr_reg_bits(ISP_SCWR_TOP_CTRL, 0, 13, 3);
diff --git a/linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc1.c b/linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc1.c
index c0bcc118..c61ee768 100644
--- a/linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc1.c
+++ b/linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc1.c
@@ -531,8 +531,8 @@ static void enable_isp_scale_new (
         sc_wr_reg_bits(ISP_SCWR_TOP_CTRL, 1, 19, 1);
     }
 
-    if ( mux_sel != 4  && wr_mif->reg_rgb_mode != 0 )
-        sc_wr_reg_bits(ISP_SCWR_TOP_CTRL, 5, 13, 3);
+    if ( mux_sel != 4  && wr_mif->reg_rgb_mode == 2 )
+       sc_wr_reg_bits(ISP_SCWR_TOP_CTRL, 5, 13, 3);
     else
         sc_wr_reg_bits(ISP_SCWR_TOP_CTRL, 0, 13, 3);
 
diff --git a/linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc2.c b/linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc2.c
index 81adb29c..f9b6eb06 100644
--- a/linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc2.c
+++ b/linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc2.c
@@ -531,7 +531,7 @@ static void enable_isp_scale_new (
         sc_wr_reg_bits(ISP_SCWR_TOP_CTRL, 1, 19, 1);
     }
 
-    if ( mux_sel != 4  && wr_mif->reg_rgb_mode != 0 )
+    if ( mux_sel != 4  && wr_mif->reg_rgb_mode == 2 )
         sc_wr_reg_bits(ISP_SCWR_TOP_CTRL, 5, 13, 3);
     else
         sc_wr_reg_bits(ISP_SCWR_TOP_CTRL, 0, 13, 3);
diff --git a/linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc3.c b/linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc3.c
index c495303d..014cae93 100644
--- a/linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc3.c
+++ b/linux_515_t7/kernel/v4l2_dev/src/platform/system_am_sc3.c
@@ -531,7 +531,7 @@ static void enable_isp_scale_new (
         sc_wr_reg_bits(ISP_SCWR_TOP_CTRL, 1, 19, 1);
     }
 
-    if ( mux_sel != 4  && wr_mif->reg_rgb_mode != 0 )
+    if ( mux_sel != 4  && wr_mif->reg_rgb_mode == 2 )
         sc_wr_reg_bits(ISP_SCWR_TOP_CTRL, 5, 13, 3);
     else
         sc_wr_reg_bits(ISP_SCWR_TOP_CTRL, 0, 13, 3);
-- 
2.25.1

